src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,136 @@
+package duke;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.List;
+import java.util.stream.Collectors;
+
+class Parser {
+
+    private TaskList taskList;
+    Parser(TaskList taskList) {
+        this.taskList = taskList;
+    }
+    String parse(String input) {
+        if (input.equals("bye")) {
+            return sayBye();
+        } else if (input.equals("list")) {
+            return returnList(taskList);
+        } else if (input.startsWith("mark")) {
+            return mark(input);
+        } else if (input.startsWith("unmark")) {
+            return unmark(input);
+        } else if (input.startsWith("delete")) {
+            return delete(input);
+        } else if (input.startsWith("find")) {
+            return find(input);
+        } else {
+            return addTask(input)[1];
+        }
```

Comment:
Out of curiousity, any reason why you chose to use _if-else_ here instead of _switch_ like below?